src/parser/mod.rs

```diff
@@ -12289,6 +12352,137 @@ impl<'a> Parser<'a> {
         }
         false
     }
+
+    /// Look for an expected keyword, without consuming it
+    fn peek_keyword(&self, expected: Keyword) -> bool {
+        match self.peek_token().token {
+            Token::Word(w) => expected == w.keyword,
+            _ => false,
+        }
+    }
+
+    /// Look for one of expected keyword, without consuming it
+    fn peek_keywords(&self, expected: &[Keyword]) -> bool {
+        for kw in expected {
+            if self.peek_keyword(*kw) {
+                return true;
+            }
+        }
+        false
+    }
+
+    fn parse_show_opt_in(&mut self) -> Result<Option<ShowStatementIn>, ParserError> {
+        let clause = match self.parse_one_of_keywords(&[Keyword::FROM, Keyword::IN]) {
+            Some(Keyword::FROM) => ShowStatementInClause::FROM,
+            Some(Keyword::IN) => ShowStatementInClause::IN,
+            _ => return Ok(None),
```

Comment:
we can return an error with e.g. `self.expected(...)` instead? since this would be a bug on our end if we hit the branch, so that the parser doesn't continue given it already consumed some tokens

Reply:
I don't think we can get to an unexpected state here because we will only get a Some if the parser consumed a FROM or an IN keyword. If it's neither, it means this show option does not exist in the statement and no other tokens were consumed, so we should just return the None.

Reply:
Ah right, yeah I think in this commit both the input and the behavior of `parse_one_of_keywords` are in sync, with code changes there's a chance that could change. One could introduce a bug like updating the input keywords and forgetting to handle the branch. It would be desirable to return an error if such ever occurs. We could make the noop branch here explicit in that case e.g?
```rust
let clause = match self.parse_one_of_keywords(&[Keyword::FROM, Keyword::IN]) {
  Some(Keyword::FROM) => ShowStatementInClause::FROM,
  Some(Keyword::IN) => ShowStatementInClause::IN,
  None => return Ok(None),
  _ => return self.expected("")
}
```